django-backend/fecfiler/reports/views.py

```diff
@@ -136,6 +136,17 @@ def amend(self, request, pk):
         report.amend()
         return Response(f"amended {report}")
 
+    @action(detail=True, methods=["post"], url_name="unamend")
+    def unamend(self, request, pk):
+        report: Report = self.get_object()
+        latest_submission = (
```

Comment:
I totally didn't think of the last upload submission needing to be restored. This is a REALLY smart way of restoring the old state.